pkg/storage/storage.go

```diff
@@ -668,3 +679,70 @@ func (s *storageModule) Monitor(ctx context.Context) <-chan pkg.PoolsStats {
 
 	return ch
 }
+
+func (s *storageModule) periodicallyCheckDiskShutdown() {
+	ticker := time.NewTicker(5 * time.Minute)
+	quit := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				log.Info().Msg("Checking pools for disks that should be shutdown...")
+				s.shutdownDisks()
+			case <-quit:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+// shutdownDisks will check the disks power status.
+// If a disk is on and it is not mounted then it is not supposed to be on, turn it off
+func (s *storageModule) shutdownDisks() {
+	for _, pool := range s.pools {
+		for _, device := range pool.Devices() {
+			log.Debug().Msgf("checking device: %s", device.Path)
+			on, err := checkDiskPowerStatus(device.Path)
+			if err != nil {
+				log.Err(err).Msgf("error occurred while checking disk power status")
+				return
```

Comment:
You aware once you return here, there is no way this routine will do it's job ever again? (unless you restart storaged) instead please log an error and try the following disk. return will exit the go routine